mdakit_membcurv/tests/test_mdakit_membcurv.py

```diff
@@ -3,10 +3,274 @@
+                                                -2.38928651e-04, 4.24992903e-02, 1.70628313e-02,
+                                                -4.85138439e-02],
+                                               [-1.94220106e-02, -2.13212988e-04, 1.05435416e-03,
+                                                1.30710748e-03, -3.49076390e-03, -1.28226749e-02,
+                                                -1.41168166e-02, -1.08870941e-02, -1.21323346e+00,
+                                                -1.55778010e-01]])}
+}
+
+
+@pytest.fixture()
+def universe():
+    u = mda.Universe(GRO_, XTC_)
+    return u
+
+
+@pytest.fixture()
+def ref_beads():
+    u = mda.Universe(GRO_PO4)
+    ref_beads = u.atoms[0:455]
+    return ref_beads
+
+
+@pytest.fixture()
+def Z_cloud(ref_beads):
+    n_cells, max_width = 10, 19
+    grid_1, grid_2, z_ref = [np.zeros([n_cells, n_cells]) for i in range(3)]
+    factor = np.float32(n_cells / max_width)
+
+    for atom in ref_beads:
+        x, y, z = atom.position / 10
+
+        cell_l = int(abs(x) * factor)
+        cell_m = int(abs(y) * factor)
+
+        grid_1[cell_l, cell_m] += z
+        grid_2[cell_l, cell_m] += 1
+
+    for i, j in it.product(range(n_cells), range(n_cells)):
+        if grid_2[i, j] > 0:
+            z_ref[i, j] += grid_1[i, j] / grid_2[i, j]
+        else:
+            z_ref[i, j] = np.nan
+
+    return z_ref
+
+
+@pytest.fixture()
+def mdtraj_po4():
+    # trajectory PO4 beads only imported using mdtraj, gone after refactoring
+    mdtraj = md.load(XTC_PO4, top=GRO_PO4)
+    return mdtraj
+
+
+@pytest.fixture()
+def md_ref_beads():
+    # reference beads using mdtraj, gone after refactoring. Select upper leaflet
+    topology = md.load(GRO_PO4).topology
+    md_ref_beads = {'upper': {'POPC': topology.select('resname POPC and index 1 to 457').astype(int).tolist()}}
+    return md_ref_beads
+
 
 def test_mdakit_membcurv_imported():
     """Sample test, will always pass so long as import statement worked"""
     assert "mdakit_membcurv" in sys.modules
+
+
+def test_md_topology():  # line 92, gone after refactoring
+    top = md.load(GRO_).topology
+    assert top.n_atoms == 18891
+
+
+def test_box_size(universe):  # line 99
+    box_size = universe.dimensions[0]
+    assert_almost_equal(box_size, 184.31013, decimal=5)
+
+
+def test_n_cells(universe):  # line 101
+    unit_width = 20
+    max_width = universe.dimensions[0] * 0.1
+    n_cells = math.ceil(max_width / unit_width * 10)
+    print(n_cells)
+    assert n_cells == 10.
+
+
+def test_dict_to_pickle():  # line 117-118
+    name = 'test_pickle_output'
+    dict_ = {'A': 1, 'B': 2, 'C': 3}
+    dict2pickle(name, dict_)
+    unpickled = pickle.load(open(name + '.pickle', 'rb'))
+    assert dict_ == unpickled
+    os.remove(name + '.pickle')
+
+
+def test_md_traj():  # line 108, gone after refactoring
+    traj = md.load(XTC_, top=GRO_)
+    assert traj.n_frames == 11
```

Comment:
Not needed, just checks that your trajectory has the length that you think it should have. This does not check _your_ code.